### PR TITLE
[DomCrawler] Improve html5Parser tests

### PR DESCRIPTION
**Pull request by victor-prdh**
| Q             | A
| ------------- | ---
| Branch?       | 6.3 
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       |  /
| License       | MIT
| Doc PR        |  /

Hi !
As mentioned by @Bilge in https://github.com/symfony/symfony/pull/49121#discussion_r1126265816, test to ensure the new html5Parser strategy wasn't the best ! So i worked a little on this subject and i come with this new proposal !

Thanks

### CODE REVIEW

File: src/Symfony/Component/DomCrawler/Tests/Html5ParserCrawlerTest.php [hunk 1]
```diff
@@ -60,6 +60,30 @@ public function testHtml5ParserNotSameAsNativeParserForSpecificHtml()
...
+    public function testHasHtml5Parser()
+    {
+        $crawler = $this->createCrawler(null, null, null, true);
+
+        $reflection = new \ReflectionClass($crawler);
+        $property = $reflection->getProperty('html5Parser');
+        $property->setAccessible(true);
+        $html5ParserValue = $property->getValue($crawler);
+
+        $this->assertEquals(\Masterminds\HTML5::class, $html5ParserValue::class, 'Html5Parser must be a Masterminds\HTML5 instance');
```
Comment (Bilge) [inline src/Symfony/Component/DomCrawler/Tests/Html5ParserCrawlerTest.php]: You probably want `assertInstanceOf` here.

File: src/Symfony/Component/DomCrawler/Tests/Html5ParserCrawlerTest.php [hunk 2]
```diff
@@ -60,6 +60,30 @@ public function testHtml5ParserNotSameAsNativeParserForSpecificHtml()
         $this->assertNotEquals($nativeCrawler->filterXPath('//h1')->text(), $html5Crawler->filterXPath('//h1')->text(), 'Native parser and Html5 parser must be different');
     }
```
Comment (Bilge) [inline src/Symfony/Component/DomCrawler/Tests/Html5ParserCrawlerTest.php:61]: Shouldn't this prior test (`testHtml5ParserNotSameAsNativeParserForSpecificHtml`) be removed now?

**Review by Bilge — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/DomCrawler/Tests/Html5ParserCrawlerTest.php (hunk 1 above)
Comment (victor-prdh) [reply src/Symfony/Component/DomCrawler/Tests/Html5ParserCrawlerTest.php]: Thanks

File: src/Symfony/Component/DomCrawler/Tests/Html5ParserCrawlerTest.php (hunk 2 above)
Comment (victor-prdh) [reply src/Symfony/Component/DomCrawler/Tests/Html5ParserCrawlerTest.php:61]: Hmm, that's not a perfect test but it still validating a certain behaviour of the domCrawler component and can be interesting to keep ?

Comment (nicolas-grekas) [reply src/Symfony/Component/DomCrawler/Tests/Html5ParserCrawlerTest.php:61]: Yes, this test case is useful to me, it tests the behavior and that's what we need also.
If the behavior of the xml parser changes, we'll know also, and this is good.

**Comment by nicolas-grekas** [comment]
Thank you @victor-prdh.

**Pull request merged**